nixos/modules/services/networking/knot.nix

```diff
@@ -142,20 +169,53 @@ let
 in {
   options = {
     services.knot = {
-      enable = mkEnableOption (lib.mdDoc "Knot authoritative-only DNS server");
+      enable = mkEnableOption "Knot authoritative-only DNS server";
+
+      enableXDP = mkOption {
+        type = types.bool;
+        default = lib.hasAttrByPath [ "xdp" "listen" ] cfg.settings;
+        defaultText = ''
+          Enabled when the `xdp.listen` setting is configured through `settings`.
+        '';
+        example = true;
+        description = ''
+          Extends the systemd unit with permissions to allow for the use of
+          the eXpress Data Path (XDP).
+
+          ::: {.note}
+            Make sure to read up on functional [limitations](https://www.knot-dns.cz/docs/3.3/html/operation.html#id21)
+            when running in XDP mode.
+          :::
```

Comment:
The link pins the 3.3 release, since the anchor jumps between release series.

Reply:
I think we'll improve that upstream.

Reply:
So we switch to this one?
https://www.knot-dns.cz/docs/3.3/singlehtml/index.html#mode-xdp-limitations
or
https://www.knot-dns.cz/docs/latest/singlehtml/index.html#mode-xdp-limitations
or perhaps extract the right major version?  :-D  (I don't have time right now)

Reply:
Did the latter.